src/main/java/io/aiven/kafka/connect/gcs/GcsSinkTask.java

```diff
@@ -103,6 +106,18 @@ public void put(final Collection<SinkRecord> records) {
         for (final SinkRecord record : records) {
             recordGrouper.put(record);
         }
+
+        if (this.config.getGcsFlushTotalMaxRecords() > 0) {
+            totalRecordCount += records.size();
+            if (this.totalRecordCount >= this.config.getGcsFlushTotalMaxRecords()) {
+                try {
+                    recordGrouper.records().forEach(this::flushFile);
+                } finally {
+                    recordGrouper.clear();
```

Comment:
mmm, I wonder why they have to clear the records on failure as well. 
Can you check what happens in that case? I assume the connector fails, is the connectortask instance recreated on restart?

Reply:
Sure, I can check it.